internal/core/application/market_price_service.go

```diff
@@ -145,122 +150,100 @@ func (m *marketPriceService) GetPrices(
-	var basePriceInRefCurrency, quotePriceInRefCurrency decimal.Decimal
-	assetPair := fmt.Sprintf("%s_%s", baseAsset, quoteAsset)
+	refPricesPerAssetPair map[string]referenceCurrencyPrice,
+) (
+	basePriceInRefCurrency decimal.Decimal,
+	quotePriceInRefCurrency decimal.Decimal,
+	err error,
+) {
+	assetPair := fmt.Sprintf("%s_%s", mktPrice.BaseAsset, mktPrice.QuoteAsset)
 	if v, ok := refPricesPerAssetPair[assetPair]; ok {
-		basePriceInRefCurrency = v.basePriceInRefCurrency
-		quotePriceInRefCurrency = v.quotePriceInRefCurrency
-	} else {
-		oneUnitOfBasePerRef, baseConvertable, oneUnitOfQuotePerRef, quoteConvertable, err :=
-			m.calcOneUnitOfAssetConvertedToRefCurrency(
-				ctx,
-				referenceCurrency,
-				baseAsset,
-				quoteAsset,
-			)
-		if err != nil {
-			return decimal.Zero, decimal.Zero, err
-		}
-
-		if baseConvertable {
-			quotePriceInRefCurrency = oneUnitOfBasePerRef
-			basePriceInRefCurrency = decimal.NewFromInt(1).Div(quotePriceInRefCurrency)
-		}
-
-		if quoteConvertable {
-			quotePriceInRefCurrency = oneUnitOfQuotePerRef.Mul(quotePrice)
-			basePriceInRefCurrency = decimal.NewFromInt(1).Div(quotePriceInRefCurrency)
-		}
-
-		refPricesPerAssetPair[assetPair] = struct {
-			basePriceInRefCurrency  decimal.Decimal
-			quotePriceInRefCurrency decimal.Decimal
-		}{
-			basePriceInRefCurrency:  basePriceInRefCurrency,
-			quotePriceInRefCurrency: quotePriceInRefCurrency,
-		}
+		return v.basePrice, v.quotePrice, nil
 	}
 
-	return basePriceInRefCurrency.Round(8), quotePriceInRefCurrency.Round(8), nil
-}
-
-//calcOneUnitOfAssetConvertedToRefCurrency returns one unit of base or quote
-//asset converted to reference currency
-func (m *marketPriceService) calcOneUnitOfAssetConvertedToRefCurrency(
-	ctx context.Context,
-	referenceCurrency string,
-	baseAsset string,
-	quoteAsset string,
-) (decimal.Decimal, bool, decimal.Decimal, bool, error) {
-	var oneUnitOfBasePerRef, oneUnitOfQuotePerRef decimal.Decimal
-	var baseConvertable, quoteConvertable = true, true
-
-	supportedFiat, err := m.raterSvc.IsFiatSymbolSupported(referenceCurrency)
+	baseAssetTicker, err := m.raterSvc.GetAssetCurrency(mktPrice.BaseAsset)
 	if err != nil {
-		return decimal.Zero, false, decimal.Zero, false, err
-	}
-	if !supportedFiat {
-		return decimal.Zero, false, decimal.Zero, false, fmt.Errorf("reference currency %s is not supported", referenceCurrency)
+		return
 	}
-
-	baseCurrency, err := m.raterSvc.GetAssetCurrency(baseAsset)
+	quoteAssetTicker, err := m.raterSvc.GetAssetCurrency(mktPrice.QuoteAsset)
 	if err != nil {
-		return decimal.Zero, false, decimal.Zero, false, err
+		return
 	}
-
-	basePerRef, err := m.raterSvc.ConvertCurrency(
-		ctx,
-		baseCurrency,
-		referenceCurrency,
-	)
-	if err != nil {
-		if err == port.ErrCurrencyNotFound {
-			baseConvertable = false
-		} else {
-			return decimal.Zero, false, decimal.Zero, false, err
+	isBaseAssetStable, _ := m.raterSvc.IsFiatSymbolSupported(baseAssetTicker)
```

Comment:
@altafan why dont we handle errors?

Reply:
not necessary to return error here. The value of such vars is used just to understand which of the 2 prices to convert to ref currency and which one will be calculated instead.
In case both are recognized as non-stable coins the service still attempts to retrieve the price in ref currency without "passing from" a stable coin.